packages/vscode-extension/src/commonlib/exchangeCode.ts

```diff
@@ -0,0 +1,42 @@
+import { Disposable, EventEmitter, Uri, UriHandler } from "vscode";
+import { ext } from "../uriHandler";
+
+export class UriEventHandler extends EventEmitter<Uri> implements UriHandler {
```

Comment:
What's this `UriEventHandler` for?

Reply:
FYI. VSCode provided `vscode.window.registerUriHandler` and `vscode.env.asExternalUri` to handle OAuth flow in codespaces. See https://code.visualstudio.com/api/advanced-topics/remote-extensions#callbacks-and-uri-handlers.

Codespaces login in Azure Account extension:
https://github.com/microsoft/vscode-azure-account/blob/main/src/login/exchangeCodeForToken.ts
https://github.com/microsoft/vscode-azure-account/blob/main/src/login/AuthProviderBase.ts#L125

Reply:
There's already a handler in `UriHandler.ts` so I guess here this one is redundant.
That's a bit different from AzAccount Ext since TTK's handler is not only for auth...

Reply:
We can use one `UriHandler` from `UriHandler.ts` here to handle the URL callback in codespace. 
The `uri.path === '/auth-complete'` statement can identify if the handler is triggered on a codespace.

Reply:
My bad, this is testing code, I will remove it. `UriHandler` is enough.